### PR TITLE
Role based views

### PR DESCRIPTION
**Pull request by iendjinn**
This is the first pass at implementing a generic method for loading permissions into apps and selecting datasets to load from based on a users permissions. I will add in the app menu based on permissions in further commits but wanted people's feedback on the suitability of this approach.

The basic idea is that when you want a datasetId to depend upon permissions you add a new case to the getDatasetId function builder in the context, that way this logic is only ever expressed in one place and the getDatasetId function can be used without worrying about what permissions a user has.

### CODE REVIEW

File: client/modules/core/actions/index.js [hunk 1]
```diff
@@ -27,18 +20,28 @@ export default {
           store.dispatch(reduxActions.authenticationError(err));
         });
     },
+    setFeedback({store}, feedback) {
+      store.dispatch(reduxActions.setFeedback(feedback));
+    },
+    setPermissions({store}, permissions) {
+      store.dispatch(reduxActions.setPermissions(permissions));
+    },
```
Comment (cbrafter) [inline client/modules/core/actions/index.js:28]: Is it worth adding a helper for checking permissions also?

**Review by cbrafter — APPROVED** [review]
Looks good so far

File: client/modules/core/actions/index.js (hunk 1 above)
Comment (iendjinn) [reply client/modules/core/actions/index.js:28]: Yes, I'll definitely be adding in a helper method. Just wanted to get the basics in place first

**Review by AshleySetter — APPROVED** [review]
Looks like a good approach to me

**Review by anthonymccaigue — APPROVED** [review]

**Review by cbrafter — APPROVED** [review]
LGTM

**Pull request merged**